### PR TITLE
Add push or replace function

### PR DESCRIPTION
**Pull request by gregorybarale**

### CODE REVIEW

File: src/pushOrReplace/pushOrReplace.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { push } from '../push/push';
```
Comment (evanliomain) [inline src/pushOrReplace/pushOrReplace.ts]: Just import `import { push } from '../push';`

File: src/pushOrReplace/pushOrReplace.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * ```
+ */
+export function pushOrReplace<T>(
+  ...replace: {
+    match: (element: T) => boolean;
+    replacement: T;
+  }[]
+): (array: ReadonlyArray<T>) => ReadonlyArray<T> {
+  return (array: ReadonlyArray<T>) => {
+    return replace.reduce(
```
Comment (evanliomain) [inline src/pushOrReplace/pushOrReplace.ts]: Prefer syntax `=> replace...`

File: src/pushOrReplace/pushOrReplace.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { push } from '../push/push';
+
+/**
+ * @module array=>array
+ */
+/**
+ * Push an element into an array, or replace an existing one if it matchs one.
```
Comment (evanliomain) [inline src/pushOrReplace/pushOrReplace.ts]: Document and test the case: `pushOrReplace({ match: x => x%2 === 0, replacement: 2 })([4, 6, 8]) === [2, 2, 2]`

File: src/pushOrReplace/pushOrReplace.ts (hunk 1 above)
Comment (gregorybarale) [reply src/pushOrReplace/pushOrReplace.ts]: Done

File: src/pushOrReplace/pushOrReplace.ts (hunk 2 above)
Comment (gregorybarale) [reply src/pushOrReplace/pushOrReplace.ts]: Done

File: src/pushOrReplace/pushOrReplace.ts (hunk 3 above)
Comment (gregorybarale) [reply src/pushOrReplace/pushOrReplace.ts]: Done

**Pull request merged**